docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/GamWithOptions.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Data;
+using Microsoft.ML.Trainers.FastTree;
+
+namespace Microsoft.ML.Samples.Dynamic.Trainers.Regression
+{
+    public static class GamWithOptions
```

Comment:
>GamWithOptions [](start = 24, length = 14)

Just an opinion, but for GAMs I'd use the other example here as well. Feels easier to understand with actual features and scenarios.

Reply:
The other sample is more of a GAM tutorial that already existed. Each model type (linear, tree, GAM, etc) has specific model inspection code (inspecting coefficients, bins, etc) that we're not covering in the API reference. The better way is to cover them in a tutorial "Making sense of GAM models" similar to the rest of cookbook, that all GAM APIs can use. 

---
In reply to: [266723489](https://github.com/dotnet/machinelearning/pull/2999#discussion_r266723489) [](ancestors = 266723489)

Reply:
Do you think we should be consistent in both examples `Gam` and `GamWithOptions`?

---
In reply to: [267022906](https://github.com/dotnet/machinelearning/pull/2999#discussion_r267022906) [](ancestors = 267022906,266723489)

Reply:
GAM sample has useful information so it's better to keep it, but I don't want to use it as a template for new GAM samples for the reasons above (we'll have two more GAM coming in classification). 

---
In reply to: [267034698](https://github.com/dotnet/machinelearning/pull/2999#discussion_r267034698) [](ancestors = 267034698,267022906,266723489)